sympy/stats/stochastic_process_types.py

```diff
@@ -759,3 +766,212 @@ def limiting_distribution(self):
+
+            if not (rsuccess <= ntrials and rsuccess >= 0):
+                raise ValueError("rsuccess <= ntrials and  ntrials, \
+                                     rsuccess must be positive Integers.")
+
+            return self.p**rsuccess * (1 - self.p)**(ntrials - rsuccess)
+
+        else:
+            if not (S(ntrials).is_Integer and ntrials > 0):
+                raise ValueError("ntrials must be a positive Integer.")
+            r = symbols('r', natural=True)
+            pmf = (self.p**r) * (1 - self.p)**(ntrials - r)
+            return DiscreteDistributionHandmade(pmf, set = FiniteSet(*range(ntrials+1)))
+
+    def _rvindexed_subs(self, expr, condition):
+        """
+        Substitutes the RandomIndexedSymbol with the RandomSymbol with
+        same name, distribution and probability as RandomIndexedSymbol.
+
+        """
+        from sympy.stats import Bernoulli
+        rvs_expr = random_symbols(expr)
+        swapdict_expr = {}
+        for rv in rvs_expr:
+            newrv = Bernoulli(rv.name, p=rv.pspace.process.p)
+            swapdict_expr[rv] = newrv
+        expr = expr.subs(swapdict_expr)
+        rvs_cond = random_symbols(condition)
+        swapdict_cond = {}
+        if condition is not None:
+            for rv in rvs_cond:
+                newrv = Bernoulli(rv.name, p=rv.pspace.process.p)
+                swapdict_cond[rv] = newrv
+            condition = condition.subs(swapdict_cond)
+        return expr, condition
+
+    def expectation(self, expr, condition=None, evaluate=True, **kwargs):
+        """
+        Computes expectation.
+
+        Parameters
+        ==========
+
+        expr: RandomIndexedSymbol, Relational, Logic
+            Condition for which expectation has to be computed. Must
+            contain a RandomIndexedSymbol of the process.
+        condition: Relational, Logic
+            The given conditions under which computations should be done.
+
+        Returns
+        =======
+
+        Expectation of the RandomIndexedSymbol.
+
+        """
+        newexpr, newcondition = self._rvindexed_subs(expr, condition)
+        return E(expr=newexpr, condition=newcondition, evaluate=evaluate, **kwargs)
+
+
+    def probability(self, condition, given_condition=None, evaluate=True, **kwargs):
+        """
+        Computes probability.
+
+        Parameters
+        ==========
+
+        condition: Relational
+        given_condition: Relational/And
+
+        Returns
+        =======
+        Probability of the condition.
+        """
+        newcondition, newgiven_condition = self._rvindexed_subs(condition, given_condition)
+        return P(condition=newcondition, given_condition=newgiven_condition,
+                 evaluate=evaluate, **kwargs)
+
+    def _pdf(self, x):
+        return Piecewise((self.p, Eq(x, 1)),
+                         (1 - self.p, Eq(x, 0)),
+                         (S.Zero, True))
```

Comment:
Is it possible to rename it as `pmf`?

Reply:
I think we can remove it, as I did't find its use now.
Also should I remove the probability_r_success and probability_rth_success ?
Though for them I will add test for which purpose they were created

Reply:
Yes remove these three methods and their tests as well. Instead add some examples of the existing API. Similar to the ones I suggested above.

Reply:
I have kept pmf as it will be used in JointDistribution